### PR TITLE
fix429EscapeE2ETimeout

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
Changes included:

Fixed Issue https://github.com/Azure/azure-sdk-for-java/issues/37419
Issue
429 escaped e2e timeout gating.
Root cause
Currently, the E2ETimeout for point operations is not sitting on top of ClientRetryPolicy, which means for each retry from ClientRetryPolicy will reset the e2e timeout timer.
Fix
For point operations, E2E timeout should be gated for all retries by client retry policy. In order to track the cancelled request diagnostics, ScopedDiagnosticsFactory is used

Fixed Issue https://github.com/Azure/azure-sdk-for-java/issues/37589
Issue
E2E timeout is not being applied for readMany (Query cases)
Fix
Apply E2E timeout for readMany query as well

Changed 429/* backoff time when retryAfter is not returned from server. For 429/3200, SDK will retry immediately. For others, SDK will retry after 100ms

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -10,11 +10,14 @@
...
+* Fixed an issue where `CosmosEndToEndOperationLatencyPolicyConfig.endToEndOperationTimeout` is not applied for `ReadMany` - See [PR 37764](https://github.com/Azure/azure-sdk-for-java/pull/37764)
+
 #### Other Changes
 * Modified the event payload when diagnostic details are traced (vis Open telemetry traces). The diagnostics can exceed the max. attribute size of 8KB. This PR will split the diagnostics and trace them in multiple events (ordered by `SequenceNumber` attribute) to ensure the full diagnostics message is available in logged events. - See [PR 37376](https://github.com/Azure/azure-sdk-for-java/pull/37376)
 * Added `sessionRetryCfg` to the diagnostic string and modified `proactiveInit` key name to `proactiveInitCfg` in the diagnostic string. - See [PR 36711](https://github.com/Azure/azure-sdk-for-java/pull/36711)
+* Modified `429` retry backoff time when `retryAfter` is not being returned from server. For `429/3200`, SDK will retry immediately, for others SDK will backoff 100ms - See [PR 37764](https://github.com/Azure/azure-sdk-for-java/pull/3764)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:20]:
```suggestion
* Modified `429` retry backoff time when `retryAfter` is not being returned from server. For `429/3200`, SDK will retry immediately, for others SDK will backoff 100ms - See [PR 37764](https://github.com/Azure/azure-sdk-for-java/pull/37764)
```

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @FabianMeiswinkel

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ServerSideOnlyContinuationFetcherImpl.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ class ServerSideOnlyContinuationFetcherImpl<T> extends Fetcher<T> {
 
     public ServerSideOnlyContinuationFetcherImpl(BiFunction<String, Integer, RxDocumentServiceRequest> createRequestFunc,
                                                  Function<RxDocumentServiceRequest,
-                                                 Mono<FeedResponse<T>>> executeFunc,
+                                                     Mono<FeedResponse<T>>> executeFunc,
```
Comment (jeet1995) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ServerSideOnlyContinuationFetcherImpl.java:26]: Nit/non-blocking: Indentation.

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/RxDocumentClientImplTest.java [hunk 3]
```diff
@@ -48,12 +48,14 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
```
Comment (jeet1995) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/RxDocumentClientImplTest.java:51]: Nit: Unused imports.

**Review by jeet1995 — APPROVED** [review]
LGTM, thanks!

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ServerSideOnlyContinuationFetcherImpl.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ServerSideOnlyContinuationFetcherImpl.java:26]: Fixed

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/RxDocumentClientImplTest.java (hunk 3 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/RxDocumentClientImplTest.java:51]: Fixed

**Pull request merged**